lib/main_micro_alp.pm

```diff
@@ -299,6 +302,18 @@ sub load_slem_on_pc_tests {
     } elsif (get_var('PUBLIC_CLOUD_UPLOAD_IMG')) {
         loadtest("boot/boot_to_desktop");
         loadtest("publiccloud/upload_image");
+        # AISTACK test verification
+        if (get_var('PUBLIC_CLOUD_AISTACK')) {
+            loadtest("publiccloud/prepare_instance", run_args => $args);
+            loadtest("publiccloud/registration", run_args => $args);
+            loadtest "publiccloud/check_services", run_args => $args;
+            loadtest("publiccloud/slem_basic", run_args => $args);
+            loadtest('publiccloud/aistack_basic', run_args => $args);
+        }
```

Comment:
As I explaiend, this code should go into the `else block, because the Download/Upload blocks are single start-stop tests.
It shall be a new test run. 
The block in next lines 313-316 not needed.

Reply:
Yes in fact, this is what I described to you (in DM), how our tests are structured: you should consider 2 tests run in sequence: 
1. HDD preparation, triggering the first `if-block` ONLY, by setting PUBLIC_CLOUD_UPLOAD_IMG(see i.e.[this](https://openqa.suse.de/tests/15751086));  
2. then trigger the ai-basic test by setting via yaml or in a VR clone cmd-line `PUBLIC_CLOUD_AISTACK=1` and also `START_AFTER_TEST=<test-name-for-hdd-preparation>` (see i.e.[this](https://openqa.suse.de/tests/15751106#settings)).
3. all eventual other similar tests will use the hdd set by n.1 and have same _start after_.

The related template settings, defined by updating a proper test group `openqa.suse.de/group_overview/<###>` yaml, will provide, for each product/flavor and each test_name, the expected settings and HDDs names.

As an example and inspiration, pls look at this [yaml](https://gitlab.suse.de/qac/qac-openqa-yaml/-/blob/master/publiccloud/updates_slemicro.yaml#L201), L#201 for (1) and L#217 for (2).

Reply:
fixed